### PR TITLE
Ensure that classes declaring @ServerRequestFilter can contain non-static fields

### LINKED ISSUE #27752: Make it possible to have non-static SecureRandom in RESTEasy Reactive filters
### Description

Declaring a `SecureRandom` instance variable in a RESTEasy Reactive filter causes a native build failure - therefore the implementations have to create it during the request processing which can be expensive.

Please investigate if the way RESTEasy Reactive filters are created can be changed to support it

CC @geoand 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Same goes for `@ServerResponseFilter`.

This is done by making the generated classes `@ApplicationScoped`, thus ensuring
that their instantiation is done lazily instead of eagerly.

Resolves: #27752

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java [hunk 1]
```diff
@@ -466,12 +476,58 @@ private static FieldDescriptor generateConstructorAndDelegateField(ClassCreator
...
+            return false;
+        } else {
+            boolean hasFieldWithoutInject = false;
+            for (FieldInfo field : fields) {
+                if (!field.hasAnnotation(Inject.class)) {
+                    hasFieldWithoutInject = true;
+                    break;
+                }
+            }
+            return hasFieldWithoutInject;
```
Comment (Sgitario) [inline independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java:525]: What about `@RequestScoped`? Can users use this annotation? If so, you will add the `ApplicationScoped` if there are fields in spite of they annotated it with `@RequestScoped`.

**Review by Sgitario — COMMENTED** [review]
So, this was only failing in native and that's why you didn't add any tests, right?

**Comment by geoand** [comment]
> So, this was only failing in native and that's why you didn't add any tests, right?

Good point, I'll add one

File: independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java (hunk 1 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java:525]: Yeah, I thought about that but was too lazy to take it into account. I guess now I have to :)

**Comment by geoand** [comment]
PR updated

**Review by Sgitario — APPROVED** [review]
lgtm now, thanks!

**Comment by sberyozkin** [comment]
Thanks very much @geoand for looking at this enhancement

**Comment by geoand** [comment]
🙏🏼

**Pull request merged**